lib/iris/tests/integration/test_grib_load.py

```diff
@@ -64,6 +65,7 @@ def test_load_time_processed(self):
                                                "time_bound.grib2")))
         self.assertCML(cubes, ("grib_load", "time_bound_grib2.cml"))
 
+    @skip('iris-grib is currently causing errors')
     def test_load_3_layer(self):
```

Comment:
I've done a bit of experimenting with this test. The error seems to be thrown when trying to realise one of the cubes in `cubes`. We get the error:
```ValueError: cannot reshape array of size 103680 into shape (360,600)```
Its worth noting that while `cubes[1]` has shape `(360, 600)` (where 360\*600 = 12960), `cubes[2]` has shape `(360,288)` (where 360\*288 = 103680).

It suggests that perhaps the cubes are at some point interfering with each other.
I tried changing the order of the cubelist to
```cubes = iris.cube.CubeList([cubes[2], cubes[0], cubes[1]])```
and I got the error:
```RuntimeError: Invalid GRIB message: /home/travis/build/stephenworsley/iris/iris-test-data/test_data/GRIB/3_layer_viz/3_layer.grib2 @ 671744```
Traceback can be found [here](https://travis-ci.com/stephenworsley/iris/jobs/246337089).
I'm still not sure if this is an Iris or Iris-grib issue.

Reply:
Interestingly, if the `CubeList` isn't reordered then the test simply fails without errors.

Reply:
When I reordered the Cubelist as:
```cubes = iris.cube.CubeList([cubes[0], cubes[2], cubes[1]])```
I got the error:
```gribapi.errors.PrematureEndOfFileError: End of resource reached when reading message```

The traceback can be found [here](https://travis-ci.com/stephenworsley/iris/jobs/246714091).

Reply:
It seems that if all the data is realised first, the test succeeds. The data must, however, be realised in the correct order:
```
cubes[0].data
cubes[1].data
cubes[2].data
```
where the data in `cubes` is realised the step before it is reordered.
If `cubes[1].data` is realised first, it will cause an error (similarly for `cubes[2]`).

Reply:
So for some reason, the most recent version of python-ecodes is 0.9.3 rather than 2.12.3 . When the travis file specifies `python-eccodes=0.9.3` then this test no longer errors.

Reply:
For reference, this issue is discussed here: https://github.com/conda-forge/python-eccodes-feedstock/issues/49
It looks like in future, a python interface will be added to eccodes so that it will replace python-eccodes and solve this version problem: https://github.com/conda-forge/staged-recipes/pull/9731